specification/powerbidedicated/resource-manager/Microsoft.PowerBIdedicated/stable/2021-01-01/autoScaleVCores.json

```diff
@@ -0,0 +1,460 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./powerbidedicated.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.PowerBIDedicated/autoScaleVCores": {
+      "get": {
+        "description": "Gets all the auto scale v-cores for the given resource group.",
+        "x-ms-examples": {
+          "List auto scale v-cores in resource group": {
+            "$ref": "./examples/listAutoScaleVCoresInResourceGroup.json"
+          }
+        },
+        "tags": [
+          "AutoScaleVCores"
+        ],
+        "operationId": "AutoScaleVCores_ListByResourceGroup",
+        "parameters": [
+          {
+            "$ref": "./powerbidedicated.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./powerbidedicated.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./powerbidedicated.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AutoScaleVCoreListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./powerbidedicated.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.PowerBIDedicated/autoScaleVCores": {
+      "get": {
+        "description": "Lists all the auto scale v-cores for the given subscription.",
+        "x-ms-examples": {
+          "List auto scale v-cores in subscription": {
+            "$ref": "./examples/listAutoScaleVCoresInSubscription.json"
+          }
+        },
+        "tags": [
+          "AutoScaleVCores"
+        ],
+        "operationId": "AutoScaleVCores_ListBySubscription",
+        "parameters": [
+          {
+            "$ref": "./powerbidedicated.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./powerbidedicated.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AutoScaleVCoreListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./powerbidedicated.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```

Comment:
>"nextLinkName": null [](start = 10, length = 20)

Same as above, are you sure that you don't want to implement pagination? See [here](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi009-always-add-x-ms-pageable-to-list-calls)

Reply:
The autoScaleVCores resource is always associated with a capacities resource. One capacity can have 0 or 1 autoScaleVCore. So there shouldn't be many autoScaleVCore resources for a tenant/subscription based on our business scenarios.

Reply:
if autoScaleVCore is a singleton - the {vCoreName} should be default or current

---
In reply to: [572330235](https://github.com/Azure/azure-rest-api-specs/pull/12244#discussion_r572330235) [](ancestors = 572330235)

Reply:
When you say "One capacity can have 0 or 1 autoScaleVCore" above - what is capacity referring to?

---
In reply to: [572577962](https://github.com/Azure/azure-rest-api-specs/pull/12244#discussion_r572577962) [](ancestors = 572577962,572330235)

Reply:
also adding a nextLinkName here does not mean you have to implement paging - it just means that your schema allows it, so that if ever you need to implement paging you don't need to make a schema change

---
In reply to: [572578316](https://github.com/Azure/azure-rest-api-specs/pull/12244#discussion_r572578316) [](ancestors = 572578316,572577962,572330235)

Reply:
"capacities" is the other tracked resource type we defined in ./powerbidedicated.json.

I wouldn't say autoScaleVCore is a singleton resource from ARM's point of view; it's a plain tracked ARM resource with basic CRUD support. Specifically, customers have read-only access to their autoScaleVCores, and our service manages autoScaleVCores in customers' subscriptions and make sure one customer's capacity has 0 or 1 autoscalevcore. Our service still need to give it a meaning name which is "AutoScaleVCore_{capacityObjectId}".

If it's not a blocking issue, I'd like to add nextLinkName settings in a new API version, and make the same change to capacities resource type.